dataset/groupby.cpp

```diff
@@ -144,23 +144,25 @@ static constexpr auto sum = [](const DataArrayView &out,
 };
 
 template <void (*Func)(const VariableView &, const VariableConstView &)>
-static constexpr auto reduce_idempotent =
-    [](const DataArrayView &out, const auto &data_container,
-       const GroupByGrouping::group &group, const Dim reductionDim,
-       const Variable &mask) {
-      bool first = true;
-      for (const auto &slice : group) {
-        const auto data_slice = data_container.slice(slice);
-        if (mask)
-          throw std::runtime_error(
-              "This operation does not support masks yet.");
-        if (first) {
-          out.data().assign(data_slice.data().slice({reductionDim, 0}));
-          first = false;
+struct wrap {
```

Comment:
Better name for `struct` than `wrap` or a comment indicating what it's achieving?